File: backend/src/main/java/com/mini/credit/enums/TypeEcartCaisse.java
package com.mini.credit.enums;

/**
 * PHASE 7: Types d'écarts caisse.
 */
public enum TypeEcartCaisse {
    DEFICIT("Déficit caisse (argent manquant)"),
    EXCEDENT("Excédent caisse (argent supplémentaire)"),
    VARIANCE_RECETTE("Variance sur recette journalière (cash != montant)"),
    VARIANCE_SESSION("Variance sur session caisse (solde physique != théorique)");

    private final String libelle;

    TypeEcartCaisse(String libelle) {
        this.libelle = libelle;
    }

    public String getLibelle() {
        return libelle;
    }
}

File: frontend/src/app/features/caisse/models/type-ecart-caisse.ts
/**
 * PATCH 9 — Types d'écart caisse.
 * Aligné sur TypeEcartCaisse.java backend.
 */
export type TypeEcartCaisse = 'DEFICIT' | 'SURPLUS';

export const TYPE_ECART_CAISSE_LABELS: Record<TypeEcartCaisse, string> = {
  DEFICIT: 'Déficit (solde physique < théorique)',
  SURPLUS: 'Surplus (solde physique > théorique)',
};
